pkg/network/tcp_transport.go

```diff
@@ -11,21 +11,21 @@ import (
 
 // TCPTransport allows network communication over TCP.
 type TCPTransport struct {
-	log      *zap.Logger
-	server   *Server
-	listener net.Listener
-	bindAddr string
-	lock     sync.RWMutex
-	quit     bool
+	log       *zap.Logger
+	server    *Server
+	listeners []net.Listener
+	bindAddrs []string
```

Comment:
Why don't you keep `TCPTransport` as is and have a number of them in the `Server`?

Reply:
Mostly because of this line:
```
		// TODO: here we need to pick up a single transporter, it will be used to
		//  deal and it doesn't matter which one.
		s.transports[0],
```
Fixed.